packages/overlay/src/OverlayBase.ts

```diff
@@ -0,0 +1,908 @@
+        if (open === this.open) return;
+        this._open = open;
+        if (this.open) {
+            OverlayBase.openCount += 1;
+        }
+        this.requestUpdate('open', !this.open);
+    }
+
+    private _open = false;
+
+    static openCount = 1;
+
+    @property()
+    placement?: Placement;
+
+    @property({ attribute: 'receives-focus' })
+    receivesFocus: 'true' | 'false' | 'auto' = 'auto';
+
+    private releaseAriaDescribedby = noop;
+    private releaseLongpressDescribedby = noop;
+
+    @query('slot')
+    slotEl!: HTMLSlotElement;
+
+    @property({ type: Number, attribute: 'tip-padding' })
+    tipPadding?: number;
+
+    @property()
+    trigger?: string;
+
+    @state()
+    triggerElement: HTMLElement | VirtualTrigger | null = null;
+
+    @state()
+    triggerInteraction?: 'click' | 'longpress' | 'hover';
+
+    @property()
+    type: OverlayTypes = 'hint';
+
+    protected wasOpen = false;
+
+    private elementResolver = new ElementResolutionController(this);
+
+    private get usesDialog(): boolean {
+        return this.type === 'modal' || this.type === 'page';
+    }
+
+    private get popoverValue(): 'auto' | 'manual' | undefined {
+        switch (this.type) {
+            case 'modal':
+            case 'page':
+                return undefined;
+            case 'hint':
+                return 'manual';
+            default:
+                return this.type;
+        }
+    }
+
+    /* c8 ignore next 12 */
+    protected async manageDialogOpen(): Promise<void> {
+        console.warn(
+            'Implement the `manageDialogOpen` method in a class extension.'
+        );
+    }
+
+    protected async managePopoverOpen(): Promise<void> {
+        console.warn(
+            'Implement the `managePopoverOpen` method in a class extension.'
+        );
+    }
+
+    protected get requiresPosition(): boolean {
+        // Do not position "page" overlays as they should block the entire UI.
+        if (this.type === 'page' || !this.open) return false;
+        // Do not position content without a trigger element, what would you position it in relation to?
+        // Do not automaticallyu position contnent, unless it is a "hint".
+        if (!this.triggerElement || (!this.placement && this.type !== 'hint'))
+            return false;
+        return true;
+    }
+
+    protected managePosition(): void {
+        if (!this.requiresPosition || !this.open) return;
+
+        const offset = this.offset || 0;
+        const trigger = this.triggerElement as HTMLElement;
+        const placement = (this.placement as Placement) || 'right';
+        const tipPadding = this.tipPadding;
+
+        this.placementController.placeOverlay(this.dialogEl, {
+            offset,
+            placement,
+            tipPadding,
+            trigger,
+            type: this.type,
+        });
+    }
+
+    protected async manageOpen(oldOpen: boolean): Promise<void> {
+        if (!this.isConnected && this.open) return;
```

Comment:
A comment here could be handy - it's not immediately clear to me why nodes that are both disconnected and open don't need to `manageOpen`, but all others do.